parsl/executors/workqueue/workqueue_worker.py

```diff
@@ -1,51 +1,43 @@
+from parsl.data_provider.files import File
+from parsl.utils import get_std_fname_mode
 import sys
 import pickle
 
 
-def check_file(parsl_file_obj, mapping, file_type_string):
-    type_desc = str(type(parsl_file_obj))
-    # Rename the file type string to the appropriate string
-    if file_type_string is None:
-        file_type_string = "<class 'parsl.data_provider.files.File'>"
-    # Obtain the local_path from the mapping
-    if type_desc == file_type_string:
-        if parsl_file_obj.filepath in mapping:
-            parsl_file_obj.local_path = mapping[parsl_file_obj.filepath]
+def load_pickled_file(filename):
+    try:
+        with open(filename, "rb") as f_in:
+            return pickle.load(f_in)
+    except pickle.PickleError:
+        print("Error while unplickling {}".format(filename))
```

Comment:
Indeed! This is related to issue #1703.  Since the previous code sometimes missed to send the appropriate errors to parsl, I did not work to include a fix for that in this pr. Do you recommend to fix it in this pr?

Reply:
Yadu, since the code changed a bit anyway, I am going to push the changes related to serializing the errors here. That way you don't have to re-read code that has changed.